src/ru/nsu/ccfit/zuev/osu/TrackInfo.java

```diff
@@ -210,6 +196,18 @@ public String getMD5() {
         return md5;
     }
 
+
+    /**
+     * Intended to be used at SongMenu reparse.
+     */
+    public void applyDifficulty(BeatmapData data) {
+
+        overallDifficulty = data.difficulty.od;
+        approachRate = data.difficulty.ar;
+        circleSize = data.difficulty.cs;
+        hpDrain = data.difficulty.hp;
+    }
```

Comment:
If this does not serve other purposes than displaying updated information at `SongMenu` during reparsing, I think it should also update other fields, as these are not the only beatmap information that can be changed. Otherwise, it appears to be a half-baked solution.